### PR TITLE
PaddlePaddle frontend - OP Conversion Part1

### PR DESCRIPTION
**Pull request by zhangYiIntel**
### Details:
 - *enable 6 op conversions for PDPD Frontend*
 - *enable test cases for op conversion*

### Tickets:
 - *53052*

### CODE REVIEW

**Comment by zhangYiIntel** [comment]
@ceciliapeng2011 @luo-cheng2021  Please also have a review

File: ngraph/test/files/paddlepaddle/gen_scripts/generate_batch_norm.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#
+# pool2d paddle model generator
+#
+import numpy as np
+from save_model import saveModel
+import sys
+
+
+def batch_norm1(name : str, x, scale, bias, mean, var, data_layout):
```
[the remark this answers is not in the document]
Comment (zhangYiIntel) [reply ngraph/test/files/paddlepaddle/gen_scripts/generate_batch_norm.py:9]: Hi. Those scripts are used to generate example models from PDPD. We utilize those generated PDPD model for op conversion test.

File: ngraph/test/CMakeLists.txt [hunk 2]
```diff
@@ -665,5 +665,6 @@ if (NGRAPH_PDPD_FRONTEND_ENABLE AND PDPD_PYTHON_OK)
     add_custom_target(pdpd_test_models DEPENDS ${OUT_FILES})
     add_dependencies(unit-test pdpd_test_models)
     add_dependencies(unit-test paddlepaddle_ngraph_frontend)
+    target_link_libraries(unit-test PRIVATE cnpy)
```
Comment (nosovmik) [inline ngraph/test/CMakeLists.txt:668]: this shall be outside of 'if' as pdpd-independent shared/op_fuzzy.cpp includes <cnpy.h>
Also please correct PDPD_FOUND_OK to paddlepaddle_FOUND on line 646

Comment (zhangYiIntel) [reply ngraph/test/CMakeLists.txt:668]: Do you mean to use `paddlepaddle_FOUND` instead of `PDPD_PYTHON_OK` ?

File: ngraph/frontend/paddlepaddle/src/op/batch_norm.cpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+namespace ngraph
+{
+    namespace frontend
+    {
+        namespace pdpd
+        {
+            namespace op
+            {
+                NamedOutputs batch_norm(const NodeContext& node)
```
Comment (jane-intel) [inline ngraph/frontend/paddlepaddle/src/op/batch_norm.cpp:16]: I see activation incorporated into the batch norm [here](https://www.paddlepaddle.org.cn/documentation/docs/zh/1.6/api_cn/layers_cn/batch_norm_cn.html#batch-norm) Am I looking at the wrong spec?

**Comment by zhangYiIntel** [comment]
Links to  PDPD Ops Spec

arg_max: https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/argmax_en.html#argmax
assign_value: https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/assign_en.html#assign
batch_norm: https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/static/nn/batch_norm_en.html#batch-norm
cast: https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/cast_en.html#cast
clip: https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/clip_cn.html#clip
concat: https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/concat_cn.html#concat

File: ngraph/frontend/paddlepaddle/src/op/batch_norm.cpp (hunk 3 above)
Comment (zhangYiIntel) [reply ngraph/frontend/paddlepaddle/src/op/batch_norm.cpp:16]: Latest Batch Norm doc is https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/static/nn/batch_norm_en.html#batch-norm
Please use PaddlePaddle doc >= 2.1

Comment (jane-intel) [reply ngraph/frontend/paddlepaddle/src/op/batch_norm.cpp:16]: References you've provided is for the Python API -- not for the operations themselves. We should not rely on it.
Here is the better place to see which attributes are included in BatchNorm https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/batch_norm_op.cc#L196

Comment (zhangYiIntel) [reply ngraph/frontend/paddlepaddle/src/op/batch_norm.cpp:16]: Thanks for your reminder, we actually rely on those PDPD cpp codes for attribute extraction. I will use them as PDPD reference links for other ops.

**Review by nosovmik — APPROVED** [review]

File: ngraph/test/files/paddlepaddle/gen_scripts/generate_batch_norm.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ilyachur) [reply ngraph/test/files/paddlepaddle/gen_scripts/generate_batch_norm.py:9]: I still have some questions about these scripts. Looks like the number of such scripts will be increased.
Who will use it?
@nosovmik What do you think about scripts?

I don't want to add service scripts which are not used in CI to the OV repository.

[the remark this answers is not in the document]
Comment (nosovmik) [reply ngraph/test/files/paddlepaddle/gen_scripts/generate_batch_norm.py:9]: For PDPD operations unit testing we had to have some small test models as there is no way to test it without real models
Initially during development we tried approach of putting already generated models to some LFS folder. But anyway we needed to keep 'generated scripts' just for reference here, so we've decided to have only scripts which will generate models during unit tests building
So, these scripts are executed and generated model + numpy inputs/outputs when you build 'unit-test' with pdpd frontend enabled

**Review by jane-intel — APPROVED** [review]

**Review by mvafin — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

File: ngraph/test/frontend/paddlepaddle/op_fuzzy.cpp [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+#include "util/test_control.hpp"
+#include <cnpy.h>
+#include "op_fuzzy.hpp"
+#include "ngraph/ngraph.hpp"
+
+using namespace ngraph;
+using namespace InferenceEngine;
+using namespace ngraph;
+using namespace ngraph::frontend;
+using TestEngine = test::IE_CPU_Engine;
```
Comment (ilya-lavrenov) [inline ngraph/test/frontend/paddlepaddle/op_fuzzy.cpp:17]: I believe in unit tests we must not perform inference. Can we just compare with referecnce ngraph function?

CC @ilyachur @zhangYiIntel @nosovmik

Comment (zhangYiIntel) [reply ngraph/test/frontend/paddlepaddle/op_fuzzy.cpp:17]: Hi, this PaddlePaddle frontend is used to convert PaddlePaddle Model to OV Ngraph Model just like ONNX Importer in OV. PaddlePaddle Framework hopes their model can run on OV CPU plugin without accuracy loss. Therefore, we use the IE_CPU_Engine here for test just like ONNX's test cases

Comment (ilya-lavrenov) [reply ngraph/test/frontend/paddlepaddle/op_fuzzy.cpp:17]: Accuracy should be covered with e2e tests, not with unit tests for ngraph

Comment (ceciliapeng2011) [reply ngraph/test/frontend/paddlepaddle/op_fuzzy.cpp:17]: I am also considering this problem. Some Op, like Deformable Conv, multiclass_nms and matrix_nms, their cpu plugin is WIP. But we can make Op conversion happen once ngraph reference function is implemented. So to use INTERPRETER instead of IE_CPU as test engine, is a good suggestion.

But there is a concern -
Have all Op implemented its reference function correctly? 
This concern comes from a real problem - PDPD yolo-box is decomposed. And tests prove that IE_CPU engine has solid accurate result in comparison to INTERPRETER.

Comment (ilyachur) [reply ngraph/test/frontend/paddlepaddle/op_fuzzy.cpp:17]: @zhangYiIntel @ceciliapeng2011 I agree with @ilya-lavrenov.

We shouldn't duplicate inference tests and mix it with other types of tests.
Please refactor these tests and just compare the graph with reference function.

Inference we can run as a part of e2e tests.

Comment (zhangYiIntel) [reply ngraph/test/frontend/paddlepaddle/op_fuzzy.cpp:17]: Thanks all for your advice. Refactor to Use INTERPRETER_Engine for op Fuzzy test
in #6614
